other-examples/confluentcloud/collector.yaml

```diff
@@ -0,0 +1,57 @@
+receivers:
+  kafkametrics:
```

Comment:
I spun this up and saw two groups of metrics sent to new relic. The first, from this `kafkametrics` receiver looks like:
```
{
    "description": "Number of brokers in the cluster.",
    "instrumentation.provider": "opentelemetry",
    "kafka.brokers": {
      "type": "gauge",
      "count": 1,
      "sum": 18,
      "min": 18,
      "max": 18,
      "latest": 18
    },
    "metricName": "kafka.brokers",
    "newrelic.source": "api.metrics.otlp",
    "otel.library.name": "otelcol/kafkametricsreceiver",
    "otel.library.version": "0.83.0",
    "timestamp": 1697223619762,
    "unit": "{brokers}"
  }
```
Notice how there is no `service.name` so entity synthesis isn't triggered and no `entity.guid` is stamped on the record.

The second comes from the prometheus receiver and looks something like:
```
{
  "description": "The approximate number of new series in this scrape",
  "entity.guid": "Mjk2NzA1MnxFWFR8U0VSVklDRXwtNTcyMDUyMjg5MjQ2NzI0Mzc5OA",
  "entity.name": "confluent",
  "entity.type": "SERVICE",
  "http.scheme": "https",
  "instrumentation.provider": "opentelemetry",
  "metricName": "scrape_series_added",
  "net.host.name": "api.telemetry.confluent.cloud",
  "net.host.port": "443",
  "newrelic.source": "api.metrics.otlp",
  "otel.library.name": "otelcol/prometheusreceiver",
  "otel.library.version": "0.83.0",
  "scrape_series_added": {
    "type": "gauge",
    "count": 1,
    "sum": 11,
    "min": 11,
    "max": 11,
    "latest": 11
  },
  "service.instance.id": "api.telemetry.confluent.cloud:443",
  "service.name": "confluent",
  "timestamp": 1697223600338
}
```
These metrics do have a service.name, which is appended automatically by the prometheus receiver based on the `job_name`, and are thus associated with an entity and stamped with `entity.guid`. However, the have a different naming convention than the metrics from `kafkametrics`: `kafkametrics` have a `kafka.*` prefix while the `prometheus` metrics aren't namespaced at all.

Question for you: is this the recommendation we want to make to customers for how we want them to configure confluent kafka metrics to land in New Relic? Seems like we want:
- Metrics to be associated with an entity, i.e. all metrics have a `service.name`
- All metrics to be named similarly, maybe with a common prefix like `kafka.*`

Reply:
For now I believe the suggestion is just to view the metrics as `kafka.*`. I believe we have a dashboard already built to handle these metrics, but I will probably make some modifications to that this week if it is off from what this collector receives.

Eventually there will be more brokers metrics available, but I'm still resolving an issue in the OTel pipeline for this. Once it merges I can look into the entities a little deeper.

Reply:
We discussed this and the [linked document](https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/collector/collector-configuration-examples/opentelemetry-collector-kafka-confluentcloud/) contains a bunch of other useful details for users on how to consume these metrics in a dashboard.